### PR TITLE
Update ASC example

### PR DESCRIPTION
**Pull request by amityadav0**
*  Add getLogicSignature function that will return logicSignature systematically 
* Update ASC example to work with GOLD ASA
* Add pyTeal file for same example

### CODE REVIEW

File: examples/deployment/src/asa-helpers.js [hunk 1]
```diff
@@ -46,3 +46,27 @@ exports.transferMicroAlgosContract = async function (deployer, fromAccount, toAc
...
+    params);
+  
+    let rawSignedTxn = algosdk.signLogicSigTransactionObject(xtxn, lsig);
+  
+    // send raw LogicSigTransaction to network    
+    let tx1 = (await deployer.algodClient.sendRawTransaction(rawSignedTxn.blob).do());
+  
+    return await deployer.waitForConfirmation(tx1.txId);
+    
+}
```
Comment (robert-zaremba) [inline examples/deployment/src/asa-helpers.js:72]: didn't we already move this helpers to `algob/src`  in a previous sprint?

File: examples/deployment/src/asa-helpers.js [hunk 2]
```diff
@@ -46,3 +46,27 @@ exports.transferMicroAlgosContract = async function (deployer, fromAccount, toAc
   })
   return await deployer.waitForConfirmation(pendingTx.txId)
 }
+
+// Transfer ASA
+exports.transferASAContract = async function (deployer, fromAccount, toAccount, amountMicroAlgos, assetID, lsig) {
```
Comment (robert-zaremba) [inline examples/deployment/src/asa-helpers.js:51]: let's rename to `transferASALsig`

**Review by robert-zaremba — CHANGES_REQUESTED** [review]

File: examples/deployment/scripts/0-gold-asa.js [hunk 3]
```diff
@@ -15,11 +15,12 @@ async function run(runtimeEnv, deployer) {
...
   const goldOwnerAccount = deployer.accountsByName.get("gold-owner-account");
   const johnAccount = deployer.accountsByName.get("john-account");
-
+  const bobAccount = deployer.accountsByName.get("bob-account")
   // activate goldOwner and john accounts
   let promises = [
     transferMicroAlgos(deployer, masterAccount, goldOwnerAccount.addr, 401000000, {note: "ALGO PAID"}),
-    transferMicroAlgos(deployer, masterAccount, johnAccount.addr, 401000000, {note: "ALGO PAID"})]
+    transferMicroAlgos(deployer, masterAccount, johnAccount.addr, 401000000, {note: "ALGO PAID"}),
+    transferMicroAlgos(deployer, masterAccount, bobAccount.addr, 1000000, {note: "ALGO PAID"})]
```
[the remark this answers is not in the document]
Comment (amityadav0) [reply examples/deployment/scripts/0-gold-asa.js:23]: ok

File: examples/deployment/scripts/0-gold-asa.js [hunk 4]
```diff
@@ -30,7 +31,8 @@ async function run(runtimeEnv, deployer) {
     //validRounds: 1002
   })
   console.log(asaInfo)
-
+  // OPT-IN for Bob Account
```
[the remark this answers is not in the document]
Comment (amityadav0) [reply examples/deployment/scripts/0-gold-asa.js:34]: ok

File: examples/deployment/scripts/2-gold-asc.js [hunk 5]
```diff
@@ -7,7 +7,7 @@ async function run(runtimeEnv, deployer) {
 
   await transferMicroAlgos(deployer, masterAccount, goldOwnerAccount.addr, 200000000, {note: "ALGO PAID"});
 
-  const ascInfo = await deployer.deployASC("3-gold-asc.teal", [],
+  const ascInfo = await deployer.deployASC("4-gold-asa.teal", [],
```
[the remark this answers is not in the document]
Comment (amityadav0) [reply examples/deployment/scripts/2-gold-asc.js:10]: ok, the task of adding modes in contract will make it more clear, i will add the difference

File: examples/deployment/src/asa-helpers.js (hunk 1 above)
Comment (amityadav0) [reply examples/deployment/src/asa-helpers.js:72]: yes, ASA helpers are moved, i will move all of the ASC, and update this too

File: examples/deployment/src/asa-helpers.js (hunk 2 above)
Comment (amityadav0) [reply examples/deployment/src/asa-helpers.js:51]: ok

File: examples/deployment/scripts/transfer/gold_sc.js [hunk 6]
```diff
@@ -1,26 +1,37 @@
...
+    // Error
+    //console.error(e);
+  }
+  // Transaction fail as John tried to send instead of GoldOwner
+  try {
+    await transferASAContract(deployer, johnAccount, bobAccount, 100, assetID, lsigJohn);
+  } catch(e) {
+    console.log('Transaction Failed - rejected by logic')
+    // Error
+    //console.error(e);
```
[the remark this answers is not in the document]
Comment (amityadav0) [reply examples/deployment/scripts/transfer/gold_sc.js:34]: Ok, done

File: packages/algob/src/lib/tx.ts [hunk 7]
```diff
@@ -159,3 +159,58 @@ export async function transferAsset (
   });
   return await deployer.waitForConfirmation(xtx.txId);
 }
+
+export async function transferMicroAlgosLsig (
```
Comment (robert-zaremba) [inline packages/algob/src/lib/tx.ts:163]: Please add doc strings to the new functions

**Review by robert-zaremba — APPROVED** [review]
Approving, but I left one more comment.

File: packages/algob/src/lib/tx.ts (hunk 7 above)
Comment (amityadav0) [reply packages/algob/src/lib/tx.ts:188]: ok

**Pull request merged**